lib/matplotlib/backends/backend_qt5.py

```diff
@@ -270,8 +264,24 @@ def _update_dpi(self):
             self.resizeEvent(event)
             # resizeEvent triggers a paintEvent itself, so we exit this one
             # (after making sure that the event is immediately handled).
-            return True
-        return False
+
+    def _update_dpi(self, dpi):
+        # Handler for screen DPI change events.
+        self._update_pixel_ratio()
```

Comment:
You don't need the _update_dpi wrapper but can instead directly connect _update_pixel_ratio, because pyqt5 detects the argument mismatch and simply drops the extra argument (Per the Qt5 docs at https://doc.qt.io/qt-5/signalsandslots.html: "a slot may have a shorter signature than the signal it receives because it can ignore extra arguments.").  This would be specifically useful for mplcairo, which would be able to just call `if hasattr(self, "_update_dpi"): self._update_dpi()` for backcompat, although another solution would be to rename _update_dpi to anything that's else.

Reply:
Yes, that works; I've remove `_update_dpi`. Note though that I want to move some of this handling into the base class; ee #19126.